### PR TITLE
Add Presets in AutoMM for Detection

### PR DESCRIPTION
**Pull request by FANGAreNotGnu**
- move val_metric from direct input of `MultiModalPredictor.__init__` to `hyperparameters`
- add presets and fix val_metric in tuto and example
- bypass the `best_score` calculation in `_top_k_average` for detection

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: multimodal/src/autogluon/multimodal/presets.py [hunk 1]
```diff
@@ -102,9 +102,96 @@ def zero_shot_image_classification():
     }
 
 
+@automm_presets.register()
+def medium_quality_faster_train_object_detection():
```
Comment (bryanyzhu) [inline multimodal/src/autogluon/multimodal/presets.py:106]: I think the preset name we agreed on is fast_inference and faster_inference.

**Review by bryanyzhu — APPROVED** [review]
LGTM with a minor comment.

**Review by sxjscience — APPROVED** [review]
LGTM

File: multimodal/src/autogluon/multimodal/presets.py (hunk 1 above)
Comment (FANGAreNotGnu) [reply multimodal/src/autogluon/multimodal/presets.py]: Just changed the naming. But in "faster" preset, I used validation loss as `val_metric`, which largely increased the training speed on small dataset (from 1474 seconds to 103 seconds on watercolor), and it didn't affect the inference speed. Comparing `faster` and `fast`, the inference fps difference is 85 vs 72, but the training time is 103 seconds v.s. 1645 seconds.

Comment (bryanyzhu) [reply multimodal/src/autogluon/multimodal/presets.py]: I see. I think we can have additional presets if that preset is likely to be used. Because each problem type has its own characteristic. Detection might care about both training and inference time, but matching maybe care more about accuracy.

File: multimodal/src/autogluon/multimodal/configs/optimization/adamw.yaml [hunk 2]
```diff
@@ -17,6 +17,7 @@ optimization:
   gradient_clip_algorithm: "norm"
   track_grad_norm: -1 # Whether to check gradient norm. We can set it to 2 to check for gradient norm.
   log_every_n_steps: 10
+  val_metric: null
```
Comment (zhiqiangdon) [inline multimodal/src/autogluon/multimodal/configs/optimization/adamw.yaml:20]: I suggest to use `validation_metric` to make it more distinguishable from the already existing `eval_metric`.

Comment (FANGAreNotGnu) [reply multimodal/src/autogluon/multimodal/configs/optimization/adamw.yaml:20]: Oh sure, I'll change this in next pr
